test/test_utils.py

```diff
@@ -112,59 +255,130 @@ def build_quadruplets(with_preprocessor=False):
+ids_quadruplets_learners_b = list(map(lambda x: x.__class__.__name__,
+                                  [learner for (learner, _) in
+                                   quadruplets_learners_b]))
+
+triplets_learners_b = [(MockTripletsIdentityBilinearLearner(), build_triplets)]
+ids_triplets_learners_b = list(map(lambda x: x.__class__.__name__,
+                                   [learner for (learner, _) in
+                                    triplets_learners_b]))
+
+pairs_learners_b = [(MockPairIdentityBilinearLearner(), build_pairs)]
+ids_pairs_learners_b = list(map(lambda x: x.__class__.__name__,
+                                [learner for (learner, _) in
+                                 pairs_learners_b]))
+# -- Supervised
+classifiers_b = [(RandomBilinearLearner(), build_classification),
+                 (IdentityBilinearLearner(), build_classification)]
 
+ids_classifiers_b = list(map(lambda x: x.__class__.__name__,
+                             [learner for (learner, _) in
+                              classifiers_b]))
+# -- Bilinear sets
+tuples_learners_b = pairs_learners_b + triplets_learners_b + \
+                  quadruplets_learners_b
+ids_tuples_learners_b = ids_pairs_learners_b + ids_triplets_learners_b \
+                      + ids_quadruplets_learners_b
+
+supervised_learners_b = classifiers_b
+ids_supervised_learners_b = ids_classifiers_b
+
+metric_learners_b = tuples_learners_b + supervised_learners_b
+ids_metric_learners_b = ids_tuples_learners_b + ids_supervised_learners_b
+
+# General sets (Mahalanobis + Bilinear)
+# -- Weakly Supervised learners individually
+pairs_learners = pairs_learners_m + pairs_learners_b
+ids_pairs_learners = ids_pairs_learners_m + ids_pairs_learners_b
+triplets_learners = triplets_learners_m + triplets_learners_b
+ids_triplets_learners = ids_triplets_learners_m + ids_triplets_learners_b
+quadruplets_learners = quadruplets_learners_m + quadruplets_learners_b
+ids_quadruplets_learners = ids_quadruplets_learners_m + \
+                         ids_quadruplets_learners_b
+
+# -- All weakly supervised learners
+tuples_learners = tuples_learners_m + tuples_learners_b
+ids_tuples_learners = ids_tuples_learners_m + ids_tuples_learners_b
+
+# -- Supervised learners
+supervised_learners = supervised_learners_m + supervised_learners_b
+ids_supervised_learners = ids_supervised_learners_m + ids_supervised_learners_b
+
+# -- Weakly Supervised + Supervised learners
+metric_learners = metric_learners_m + metric_learners_b
+ids_metric_learners = ids_metric_learners_m + ids_metric_learners_b
+
+# -- For sklearn pipeline: Pair + Supervised learners
+metric_learners_pipeline = pairs_learners_m + pairs_learners_b + \
+                         supervised_learners_m + supervised_learners_b
+ids_metric_learners_pipeline = ids_pairs_learners_m + ids_pairs_learners_b +\
+                             ids_supervised_learners_m + \
+                             ids_supervised_learners_b
```

Comment:
I forgot the rationale for this list. What is the criterion for inclusion?
My first thought would be that it needs to implement the `transform` method. In which case bilinear learners should not be part of this. But at the same time we include pair learners and not others (although they can all implement transform as long as they are Mahalanobis)
Are you able to clarify? And if so, add a comment to explain this explicitly